code/messaging/src/test/java/com/adobe/marketing/mobile/messaging/ContentCardSchemaDataUtilsTest.kt

```
@@ -0,0 +1,273 @@
+        val result = ContentCardSchemaDataUtils.createAepDismissButton(dismissMap ?: emptyMap())
+        assertNull(result)
+    }
+
+    @Test
+    fun `test createAepDismissButton with circle style`() {
+        val dismissMap = contentCardMap[MessagingConstants.ContentCard.UIKeys.DISMISS_BTN] as? Map<String, Any>
+        val map = dismissMap?.toMutableMap()?.apply {
+            put(MessagingConstants.ContentCard.UIKeys.STYLE, MessagingConstants.ContentCard.UIKeys.CIRCLE)
+        } ?: emptyMap()
+        val result = ContentCardSchemaDataUtils.createAepDismissButton(map)
+        assertNotNull(result)
+        assertEquals(R.drawable.cancel_filled, result?.drawableId)
+    }
+
+    @Test
+    fun `test createAepDismissButton with unsupported style`() {
+        val dismissMap = contentCardMap[MessagingConstants.ContentCard.UIKeys.DISMISS_BTN] as? Map<String, Any>
+        val map = dismissMap?.toMutableMap()?.apply {
+            put(MessagingConstants.ContentCard.UIKeys.STYLE, "unsupported_style")
+        } ?: emptyMap()
+        val result = ContentCardSchemaDataUtils.createAepDismissButton(map)
+        assertNull(result)
+    }
+
+    @Test
+    fun `test getTemplate with valid content card`() {
+        val contentCardSchemaData = mock(ContentCardSchemaData::class.java)
+        val propositionItem = mock(PropositionItem::class.java)
+        val proposition = mock(Proposition::class.java)
+
+        Mockito.`when`(contentCardSchemaData.content).thenReturn(contentCardMap)
+        Mockito.`when`(contentCardSchemaData.meta).thenReturn(metaMap)
+
+        Mockito.`when`(propositionItem.proposition).thenReturn(proposition)
+        Mockito.`when`(proposition.uniqueId).thenReturn("testId")
+
+        contentCardSchemaData.parent = propositionItem
+
+        val result = ContentCardSchemaDataUtils.getTemplate(contentCardSchemaData)
+        assertTrue(result is SmallImageTemplate)
+    }
+
+    @Test
+    fun `test getTemplate with missing title`() {
```

nit: Can we also have a ``test getTemplate with empty title`

---

We can also add a `test getTemplate with missing metadata map` which should throw an exception